solutions/security-enforced/variables.tf

```diff
@@ -0,0 +1,375 @@
+##############################################################################
+# Input Variables
+##############################################################################
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API key to deploy resources."
+  sensitive   = true
+}
+
+variable "existing_resource_group_name" {
+  type        = string
+  description = "The name of an existing resource group to provision resource in."
+  default     = "Default"
+  nullable    = false
+}
+
+variable "prefix" {
+  type        = string
+  nullable    = true
+  description = "The prefix to be added to all resources created by this solution. To skip using a prefix, set this value to null or an empty string. The prefix must begin with a lowercase letter and may contain only lowercase letters, digits, and hyphens '-'. It should not exceed 16 characters, must not end with a hyphen('-'), and can not contain consecutive hyphens ('--'). Example: prod-0205-cos. [Learn more](https://terraform-ibm-modules.github.io/documentation/#/prefix.md)."
+
+  validation {
+    # - null and empty string is allowed
+    # - Must not contain consecutive hyphens (--): length(regexall("--", var.prefix)) == 0
+    # - Starts with a lowercase letter: [a-z]
+    # - Contains only lowercase letters (a–z), digits (0–9), and hyphens (-)
+    # - Must not end with a hyphen (-): [a-z0-9]
+    condition = (var.prefix == null || var.prefix == "" ? true :
+      alltrue([
+        can(regex("^[a-z][-a-z0-9]*[a-z0-9]$", var.prefix)),
+        length(regexall("--", var.prefix)) == 0
+      ])
+    )
+    error_message = "Prefix must begin with a lowercase letter and may contain only lowercase letters, digits, and hyphens '-'. It must not end with a hyphen('-'), and cannot contain consecutive hyphens ('--')."
+  }
+
+  validation {
+    # must not exceed 16 characters in length
+    condition     = length(var.prefix) <= 16
+    error_message = "Prefix must not exceed 16 characters."
+  }
+}
+
+variable "name" {
+  type        = string
+  description = "The name of the Databases for MySQL instance. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<name>` format."
+  default     = "mysql"
+}
+
+variable "region" {
+  description = "The region where you want to deploy your instance."
+  type        = string
+  default     = "us-south"
+}
+
+variable "existing_mysql_instance_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of an existing Databases for MySql instance. If no value is specified, a new instance is created."
+}
+
+variable "mysql_version" {
```

Comment:
sync with Redis desc, same as in fully-configurable, check desc of variables in Redis and default values